### ISSUE TITLE
SIM UI: Tutor Function

### ISSUE THREAD

**Issue by jamesmrollins** [lzim/teampsd#775]
1.  **Design, develop, test and implement a tutorial function that orients new users to the SIM U.** Use the same concept as Lucid, where a user can toggle the function on or off as desired.  The tutor would be default "on" when the user logs in for the first time. This function would be available in the Sim UI and the MTL Demo Sim UI. Need some ideas for an icon for the button.

![image](https://user-images.githubusercontent.com/30132017/67448081-d22eeb80-f5ca-11e9-8d36-a6b80ce9cb66.png)

**2. Click on the right:**
a. **Projects** - assign to **issue_tracker.**
b. **Assignee** - assign relevant team members.
c. **Labels**- review and select all potential _MTL_ dependencies.
d. **Milestones** - select the dependent milestone.

-----------------------------------------------------------------------------------------
**Workgroups leads**
**Identify Constraints (check all that apply):**
- [ ]   Capacity
- [ ]   Cost
- [ ]   Schedule
- [ ]   Interference
- [ ]   Capability
- [ ]   Other (describe)

_Add estimated person-hours to complete:_ 40 hours for web and tablet,. 30 hours for admin panel that allows customization of tutor prompts.

**Edit the due date (if necessary)**

**Comment by jamesmrollins** [comment]
@lzim @staceypark @ritahitching @branscombj @dlkibbe @dlounsbu 
Please find the attached tutorial table.  The table is organized by Page, then Section and Feature.  Each feature contains a tutorial message.  I thought we could review in table form and refine the messages here.  If we feel it is, at least, a 90% complete listing I can refer to DEV for further estimation.  I think I will see if I can create an admin page so we can adjust the messages as life goes on. If we want his in place soon, I really need this feedback by next week Tuesday at the latest.  Thank you, James

[tutorial_table.docx](https://github.com/lzim/teampsd/files/3786075/tutorial_table.docx)

**Comment by ritahitching** [comment]
@jamesmrollins attached some very minor comments directly on the tutoral_table document. thanks, Rita
[tutorial_table_rh_edits_2019_11_06.docx](https://github.com/lzim/teampsd/files/3825329/tutorial_table_rh_edits_2019_11_06.docx)

**Comment by branscombj** [comment]
I made a few more edits and comments on Rita's version.
In general, I think you've got the right type and scale of content. I seem to lean toward slightly more detailed descriptions, although I would eliminate the "clicking on this icon will" parts (just personal style).

[tutorial_table_rh__jb_ edits_2019_11_08.docx](https://github.com/lzim/teampsd/files/3825675/tutorial_table_rh__jb_.edits_2019_11_08.docx)

**Comment by jamesmrollins** [comment]
@branscombj @ritahitching thanks for your feedback!  I will review and see if I can move this project to the next step (estimation).  Thank you, James

**Comment by jamesmrollins** [comment]
@hirenp-waferwire  I need an estimate of the below based on the tutorial_table.docx provided here 

> @lzim @staceypark @ritahitching @branscombj @dlkibbe @dlounsbu
> Please find the attached tutorial table. The table is organized by Page, then Section and Feature. Each feature contains a tutorial message. I thought we could review in table form and refine the messages here. If we feel it is, at least, a 90% complete listing I can refer to DEV for further estimation. I think I will see if I can create an admin page so we can adjust the messages as life goes on. If we want his in place soon, I really need this feedback by next week Tuesday at the latest. Thank you, James
> 
> [tutorial_table.docx](https://github.com/lzim/teampsd/files/3786075/tutorial_table.docx)

Also, below are some concepts to help with the estimate.  A more formal design will be provided after the estimate process.

1.  This is a similar function provided by the Lucid Meetings website.  We want to emulate this concept, however, not exactly copy it. 

![image](https://user-images.githubusercontent.com/30132017/68705923-cab48f80-0543-11ea-930b-eaea62f5cab4.png)

2.  Our button design would look similar to below. However, we are not settled on the design nor the placement yet.

![image](https://user-images.githubusercontent.com/30132017/68706341-7d84ed80-0544-11ea-9bfa-440a15cd038d.png)

3.  The tutorial_table.docx contains information regarding what tutorial balloons we would like to appear to describe a particular feature on a web page.  This should support your estimate by page, by feature.

**Comment by jamesmrollins** [comment]
@lzim @staceypark @ritahitching  Will take about 37 hours to develop the tutorial function for tablets and PC browser pages.  Additional 25 hours for smartphone format.  Recommend we hold off on smartphone development until future.

**Comment by staceypark** [comment]
@ritahitching @staceypark will look at tutorial concept today.
@lzim please let @jamesmrollins know if we can move forward on this design. I also think it makes sense to hold off on phone tutorial portion for future_release

**Comment by lzim** [comment]
I really need more time before I can approve this, unfortunately.

Let's try to get that meeting from last Friday rescheduled w/@jamesmrollins to talk about both #775 and #339 

@staceypark I do think this will be helpful sooner rather than later based on seem fo feedback I've gotten from others. It's a huge usability boost 🚀

**Comment by lzim** [comment]
Decisions from our Live Meeting:

1. Will call this the "Tutor" icon. 
2. Will live with other navigation bars, including the pop-ups.
3. Confirmed that once dismissed the tutor will stay dismissed, even if you log-off and log back in.
4. Tutor displays on first log-in and on new releases.

**Comment by staceypark** [comment]
Icons in the Header bar (Home, Play, Chat, Help, Logoff) and "i" information icon in the experiment sliders need to be higher resolution.

**Comment by lzim** [comment]
Hi @jamesmrollins and @staceypark 

@dxlouie is willing to help test functionality for the tutor 💻 when it's available in TEST.

Thanks to you all! 
Happy Friday 🍂

**Comment by jamesmrollins** [comment]
Hi @lzim and @staceypark I am happy to hear that @dxlouie is willing to help us through user testing!

<< Here is a link to an initial Tutor XD protype >> https://xd.adobe.com/view/94fca79c-5231-441c-628d-a6f817f64ad5-e2cd/  

** Note**  Be sure to go through the prototype more than once, sometimes it is a little jerky the first time through. 

**Design Notes**
1.  When a user comes into a web page for the first time, the Tutor will be active and call itself out to the user's attention.  When the user clicks the Tutor button, the tutor will continue to point out features in the section.
2.  I thought we might try a tutor button at the section level, instead of the page level.  That enables the user to select information they need by section, rather than having to cycle through elements of an entire page.  May be too overwhelming.  
3.  Explanations are short and sweet and focus on functionality and in some cases, identify what things are.  Each stop is about 5 seconds.  It can be sped up, although I don't recommend it.  
4.  Getting to know XD pretty well.  We might consider building the prototype here, having @dxlouie provide feedback, before sending to DEV for coding and then movement to TEST.  
5. For another run through, you can click on the Tutor button at the upper left of the main diagram section and it will cycle again.

**Comment by dxlouie** [comment]
Hi @lzim @staceypark @jamesmrollins is the adobe XD version the one you wanted me to user test?

I took a peek at the XD link and the tutor mode looks nice!

My main issue is that, as a user, I don't know how the sections interact. For example, how do the stock and rate indicator influence each other? Explaining the basics of the process (new patients go in here, as they are seen the stock of avail appts goes down, this causes rescheduling rate to increase etc) would be helpful. 

In my experience with highly complex, data-rich games/programs that have steep learning curves, the most useful feature I've found is the hover-over tooltip. This is already implemented for some variables as pointed out in the last page of the tutor, but would be really helpful for the Reveal Complexity options, and the Display Patient Services options. I can play around with the buttons all I want, but if they don't seem to have an effect on the system, or if I don't understand their effect on the system, then I can't learn how the system is operating.

**Comment by ritahitching** [comment]
**User Feedback of SimUI** 
- Novel users asked for more guidance on the causality of buttons and levers, how sections interact to facilitate an understanding of the impact of changes in one section of the model to another
- From the feedback a mockup tutorial was developed in an interactive, pointer concept way, to explain the header buttons 
- Full model tutorial is already available, the new mockup should  function as  compliment to the tutorial already created rather than a replacement 
- Facilitators are necessary - the core model of mtl is co-facilitation / team base learning experience, will be explaining the sections of the SimUI and systems thinking.  
- The mockup created may be helpful to the mtl.how/demo or clinicians using it between sessions, but it should not be a stand alone concept, as the principle is about team facilitation
- Focus on a technical orientation of the buttons/levers on the SimUI to complement existing resources, to call attention to existing resources and teachers
- Perhaps relabel as **"orientation** rather than **tutorial**

**Possible Directions of the orientation tutorial**
- Decision by @lzim what the 'tutorial/orientation' should look like and the depth of scope
- Is there a way to integrate co-facilitators tutorial with the feedback from the users in asking for guidance on interaction between buttons and levers?
- Possible marriage between the full model tutorial and the orientation tool - to scale up for sel-learning

**- Next grant team-facilitation vs self-facilitation** - we are asking for learners to do independent learning and the orientation tutorial may be a good reminder or compliment during individual self learning 

@staceypark @lzim @dlounsbu @branscombj

**Comment by ritahitching** [comment]
James, David and Jane discussed the feedback from dxlouie. Existing resources include:

* i information defining variables throughout the sim UI

* Help information that explains much more about how to read the diagram, run a sim, etc etc

* Facilitation for team users

* Videos for team users

* See guides with gifs for team users



I'm not sure more is needed for team users.

Question is: What more, if anything, is needed for demo users?

**Comment by staceypark** [comment]
Next steps for @jamesmrollins discussed during 1/7 meeting:
- Every "section" (blue bar header) would have a tutor for that section (e.g., "If you have any questions, use the "Main Model Diagram Tutor").

- Red animation would be in order like someone walking you through step by step on where to click, consistent with the order in the session guide and session video.

- Add small icons within the tutor that link to the relevant session guide and video for further instructions, which will be within the red pop up 

- We will stick with the red color for the tutor function.

James will have the tutor prototype ready review by the  QIICs at the Wednesday 1/22 QIIC at 9:30-11:30am Pacific

**Comment by branscombj** [comment]
@james Question about where the user's pointer needs to be to a) launch the tutor and b) expand/collapse the section. I was glad when I learned that you don't have to click precisely on the triangle in a blue bar to expand or collapse it; you could click anywhere in the blue bar. With the addition of the tutor icon in the blue bar, what changes? My hope would be that you could still click anywhere Else in the blue bar for expand/collapse and only on the tutor icon gets you the tutor.

**Comment by jamesmrollins** [comment]
@branscombj I will check it out, but we should be able to maintain the functionality you desire.  James

**Comment by jamesmrollins** [comment]
_Moving information from Basecamp to GitHub for record._

**Tutor Function Purpose**
 The purpose of the Tutor function is to provide guidance on sections/features within the SimUI, with directions to video or section guides to learn more outside of the SimUI. The Tutor function will likely be very helpful complexity is added within the  "Build Phase" sessions of mtl.

**Use Case**
Users forget and want a quick reminder, separate from the guides and videos. The overall objective of the Tutor was to help within each section of the SimUI.  The Tutor should have the option to open and close anywhere in the SimUI. Updates to software as new releases happen.

**Concerns by QIICS**
The speed of the animation seems too fast, particularly if users want to be able to follow/read the content of the information/guidance within the boxes.  DECISION: Ask James for a a mock up of the Tutor function that includes a:
1) Hover function to see more with icons for video and guides links. 
**Need clarification about this. Hover function is difficult to make 508 compliant.**
2) Quick start 'in-order tutor' of the order of steps to perform a particular task. 
**Consistent with procedures outlined in session 5 and 6 in tutorials, correct?**
3) Option for the 'Tutor' function to be available at the users request not offered by default. 
**So how are they supposed to find out about the tutor in the first place?  Recommend a tutor function highlight upon first login that can be dismissed.**
4) Independent Tutor functions for the Blue Bars
**Check**
5) Make font as large as possible within the instruction boxes, without enlarging the size of the boxes.
**I'll try 😁**
6) Slow the speed at which the Tutor function displays. 
**Check**

We will review Wednesday Jan 22. - Comments on the Tutor function tracked in GitHub #775

**Comment by jamesmrollins** [comment]
@staceypark @ritahitching @lijenn @lzim @branscombj @dlkibbe @dlounsbu   This link https://xd.adobe.com/view/4e9028fe-d44b-43a5-76c3-f30c422e0bf3-38a1/  is a prototype tutor function for the home page, consistent (for the most part) with training session 5. I will use this link to facilitate our discussion tomorrow.  Everyone is encouraged to click the link and experiment.  

Thanks, James

**Comment by jamesmrollins** [comment]
Hi @dlkibbe @branscombj !  Please find the attached table of proposed Tutor prompts. Your edits are welcome. I've included what I think are appropriate links to MTL training. 

[tutor_pop_up_scripts_01_23_20.docx](https://github.com/lzim/teampsd/files/4106036/tutor_pop_up_scripts_01_23_20.docx)

**Comment by jamesmrollins** [comment]
Hi Everyone!  Here is the updated table and the link to the prototype.  I am currently having DEV do some work break down and estimating. Hope to have in place by our meeting.

**Updated Tutor Scripts Table**
[tutor_pop_up_scripts_01_26_20.docx](https://github.com/lzim/teampsd/files/4115382/tutor_pop_up_scripts_01_26_20.docx)

**Updated Tutor Prototype**  _Beware, there are some dead ends.  If that happens, refresh the link and start at the top menu._
https://xd.adobe.com/view/c63dc602-487c-4763-5414-6a9ee9c647c6-b00c/

**Comment by branscombj** [comment]
@jamesmrollins @dlkibbe I have made suggestions in track changes; mainly cutting words for brevity. [tutor_pop_up_scripts_01_26_20.jb.docx](https://github.com/lzim/teampsd/files/4124799/tutor_pop_up_scripts_01_26_20.jb.docx)

**Comment by dlkibbe** [comment]
Thanks @jamesmrollins and @branscombj for all the work on this file. I have made some edits and added a couple of comments for your consideration.
[tutor_pop_up_scripts_01_26_20.jb.dk.docx](https://github.com/lzim/teampsd/files/4128625/tutor_pop_up_scripts_01_26_20.jb.dk.docx)

**Comment by branscombj** [comment]
Great, @dlkibbe ! 
@jamesmrollins I have accepted changes into a clean document that represents Debbie's and my combined edits. We should ask the team (HQ & QIICs) to review while in the sim UI to make sure everything looks good in actual practice. Most of my edits were made only referring to the Word table and there could be some nuances relative to live use that should be fixed. 
[tutor_pop_up_scripts_01_26_20.jb.dk.jb.docx](https://github.com/lzim/teampsd/files/4128899/tutor_pop_up_scripts_01_26_20.jb.dk.jb.docx)

**Comment by jamesmrollins** [comment]
Thanks @branscombj @dlkibbe , for the edits!  I have to present the tutorial function again to the QIICS in a few hours, so I may not be able to update these in time.  But I will try and get as far as I can.  ~James

**Comment by jamesmrollins** [comment]
FOR RELEASE DECISION:

- [x]  **Tutor Prompts table editing completed**

- [x] **DEV Estimate Complete.**  40 DEV Hours for Tutor prompts as specified in prototype, for CC, MM, AGG, PSY and SP.  No main diagram builds included in first sprint. 30 DEV Hours for admin panel to update tutor prompt text. Estimated delivery to TEST by 2/17.

- [ ]  **QIICS Review**
  
**Release?**
- [ ]   Yes

- [ ]  No

**Comment by branscombj** [comment]
@jamesmrollins @staceypark @dlkibbe  Is it true that we will be able to tweak text for these relatively simply in the future if we need to? It seems like the basic design, functionality, guiding principles about this feature are pretty much agreed upon and that maybe any further, minor text edits shouldn't hold up PROD?

**Comment by jamesmrollins** [comment]
@branscombj @dlkibbe @staceypark  I agree that it shouldn't.  Whether or not it is editable, is dependent on whether the PI agrees to the additional 30 hours of DEV time for the admin panel.  However, the text you provided is clean and I think will stand up to scrutiny.  I am editing the prototype as we speak 😁

**Comment by ritahitching** [comment]
Notes for MTL Launch Meeting Jan 29 

**Tutor Function**
- Tutor Function development - 40 hours
-  Admin Panel development for tutor function - 30 hours
- These two items are independent of each other and can be developed separately.
- These two items are independent of each other and can be developed separately.
 
**Action Items: James** 
James: Will incorporated Jane & Debbie's comments on the table of suggested text prompts for the Tutor Function.
James: will add a link to the relevant video module with a video icon and change the link icon to say "guide" in every Red Pop up box (Issue #775)

 - Tutor prompt design has now included a link button that takes users to the github guide (link can be linked to specific session/video though)

- LZ wants a separate video icon that will take you to the video tutorial, video should not just take you to the right guide, but take you to the proper module

**Red Tutor Boxes:**
Links can be adjusted on the fly.
Ideally we will have a link to the guides (will have "guide" as verbage) and an additional link that will link to the videos
Small modifications will be made from UI perspective with regards to repsonsiveness of boxes and pop ups
Not sure if possible to have it auto scroll. Concern if not would be that user could get stuck if they don't see the need to scroll

**Output and Text Section Icon Changes**
In the "Outputs and Text" section, James will move the "Expand" icon to be in the blue bar between the tutor icon and the "X" close window icon and add a "T" in the icon to signal that this will lead you to the "Text" section.
In the "Text" section, James will change the left icon in the blue bar to the letter "T" to match the "T" being added in the "Expand" icon in the "Outputs and Text" section (instead of the current graph icon).
We will keep the "Expand" icon in the "Text" section the way it currently is without the "T", since clicking on this will lead the learner to both the Text, Outputs, and more.

 

**Additional Tutor Prompts needed by Module**

In the Control Panel of the Outputs and Text (expanded) Section, James will add a red Tutor prompt to explain the different experiment views. For example for PSY, this includes the "Sankey" diagram. For Agg and MM, this includes "Compare Experiments" and "Compare Services".

Proposed tutor language for AGG from Jane:

Control Panel lets you choose which results to compare: results for various services for the current experiment or results for different experiments for a selected service

For SP, we may need an additional Tutor function (maybe in the Timeline section) that explains the different views.

Note: This will increase the cost estimate for the Tutor function.

**Auto-scrolling issues** 

James will propose a possible workaround to prevent the Tutor boxes being off screeen in the absence of auto-scrolling.

**Comment by jamesmrollins** [comment]
**@branscombj @dlkibbe Here is an updated table that I hope captures the nuances between the models.** Also added your suggested tutor prompts offered during the meeting.  I also ad-libbed one following the same style. Please take a look. I added a column for videos, but don't know about video links.  Can you put in those links? I will share with DEV tonight to update the estimate.

[tutor_pop_up_scripts_01_29_20.jb.dk.jb.jmr.docx](https://github.com/lzim/teampsd/files/4130947/tutor_pop_up_scripts_01_29_20.jb.dk.jb.jmr.docx)

**@lzim Here is an updated Icon set for the Tutor controls.**

![image](https://user-images.githubusercontent.com/30132017/73401721-88783000-42a0-11ea-85d1-bb624da0cb62.png)

**Here is an Updated Expand to Text Icon.**

![image](https://user-images.githubusercontent.com/30132017/73404131-cfb4ef80-42a5-11ea-9788-bf935a4bd82e.png)

**And here is the greyed out Select Team Data icon always displayed,** so the tutor prompt can pick up on it. Is currently hidden when "Join a Current Session" is selected.

![image](https://user-images.githubusercontent.com/30132017/73403970-70ef7600-42a5-11ea-8263-7b93d0f6f490.png)

**Comment by staceypark** [comment]
Including recommendations and decisions from 2/22 meeting for tracking:
1. @jamesmrollins will build out a table with all the tutor prompts for @branscombj and @dlkibbe review

2. The "hover" function suggested for the Sim UI Tutor is not 508 compliant. The computer cannot "read" or recognize a hover function for folks that are using assistive technology. We use the "i" information schema as this is 508 compliant and "readable" by assistive technology for the visually impaired. @jamesmrollins will mockup a tutor pop-up to explain what the PLDR Button does, as a possible work around to a "hover" function explaining the "Community" button. 

3. For the Session section, in order to direct user correctly for their session needs, the tutor prompts will follow the order below:
- Click on Tutor button
- Display pop-up that says "Join Current Session or Start a New Session"
- Follow-up with dialogue based on answer to above
@jamesmrollins will make a grayed out "Select Team Data" display button on the Session tile so the user can see that there is an additional step in "Start a New Session" 

4. Change the "PLDR" button to either say "Community" or be similar to the MTL logo with "Community" under the red arrow

5. Outgoing Messages from "Community" Button - @jamesmrollins will add in the ability to customize the subject line ins the PLDR management dashboard ("Outgoing Message Management" system). Recommendations on subject line: MTL Community - Idea; MTL Community - Like, etc.

6. Alert Messages: There can only be one alert message active at a time. Once the user dismisses it, it will not come back up. We can change this function if this does not make sense.

7. News Flash: 
- Unlike the "Alerts" function, the News flash does not automatically pop up. A user would need to click on the "News" icon in the Nav bar, which would provide a pop up with all "active" News (active, meaning we have chosen to turn the News flash "on" in the backend). 
- The "News" icon in the Nav bar includes a red number alert that notifies users of all new "News" they have not viewed yet. Once they have clicked on the "News" icon once, the red alert will disappear, but the user can click on the "News" icon at any time to see any current issues.

8. James will talk to DEV to scope out a function/dashboard for updating the Tutor pop ups and update the Workgroup leads during the Monday 1/27 meeting on feasibility with timeline for this function

**Comment by lijenn** [comment]
@jamesmrollins Not sure if we want to put the video's on YouTube due to the possibility of YouTube not addressing video buffering issues on BrightCove. 

Here are the links to the videos on BrightCove. Could give the BrightCove links a try since you are testing soon.  https://github.com/lzim/teampsd/issues/206

**Comment by staceypark** [comment]
@jamesmrollins following up on @lijenn's last comment.
Please use #206 to populate the video links throughout tutor.

**Comment by branscombj** [comment]
@jamesmrollins Here is a revised table. It is not in track changes (I accepted all of your previous edits.) I highlighted in yellow anything I edited. I added a few notes in my voice in red font.

I changed which session guide to attach in some places and completed that last column.
For the videos, I put the full list of 5 links (one for each module) in the Video Links column at the first mention of that session number. Elsewhere I just listed in the video links column which session number (should match the number in the Guide link!)

You will need to specify the correct module link for the tutor in each of those different modules, I fear. Ugh.

Let me know if questions!

[tutor_pop_up_scripts_01_29_20.jb.dk.jb.jmr (1).docx](https://github.com/lzim/teampsd/files/4309101/tutor_pop_up_scripts_01_29_20.jb.dk.jb.jmr.1.docx)
